### PR TITLE
fix(Scrollock): Allow scrolling on mobile devices when Header or Autosuggest overflow the screen size

### PR DESCRIPTION
**Pull request by kosanna**
Given I'm a user on iPhone I can't scroll within the list.

Issue: if header menu or Autosuggest list overflows the screen and screen is locked with Scrolllock component, scroll events work on inner containers, but touch scroll don't.

This PR fixes it.

### CODE REVIEW

File: react/Autosuggest/Autosuggest.js [hunk 1]
```diff
@@ -81,14 +81,16 @@ export default class Autosuggest extends Component {
       }
     };
 
+    const suggestionsContainer = document.getElementById(containerProps.id);
```
Comment (mengtzu) [inline react/Autosuggest/Autosuggest.js]: You have a guard on the menu version of this, is it not necessary here?

Comment (kosanna) [inline react/Autosuggest/Autosuggest.js]: I don't get you... ?

Comment (mengtzu) [inline react/Autosuggest/Autosuggest.js]: The `typeof document !== 'undefined' &&`

Comment (kosanna) [inline react/Autosuggest/Autosuggest.js]: It's in renderSuggestionsContainer method so I don't need it here, technically....but that's a good point, need to check

Comment (kosanna) [inline react/Autosuggest/Autosuggest.js]: you're totally right

File: react/Autosuggest/Autosuggest.js [hunk 2]
```diff
@@ -81,14 +81,16 @@ export default class Autosuggest extends Component {
       }
     };
 
+    const suggestionsContainer = typeof document !== 'undefined' && document.getElementById(containerProps.id);
```
Comment (michaeltaranto) [inline react/Autosuggest/Autosuggest.js]: Not sure i'm keen on having every consuming component needing to implement this. Feel like we should make a higher order component that implements scroll lock and makes `ref` a required property. Then we can resolve the dom node from the ref internally to the HOC. Thoughts?

Comment (kosanna) [inline react/Autosuggest/Autosuggest.js]: Might be tricky as you probably don't know which area should be scrollable.

Comment (kosanna) [inline react/Autosuggest/Autosuggest.js]: Got ya, doing

File: react/Autosuggest/Autosuggest.demo.js [hunk 3]
```diff
@@ -63,7 +63,7 @@ export default {
     id: 'jobTitles',
     label: 'Job Titles',
     autosuggestProps: {
-      suggestions: ['Developer', 'Product manager', 'Iteration manager', 'Designer'],
+      suggestions: ['Developer', 'Product manager', 'Iteration manager', 'Designer', 'UX', 'Cleaner', 'HR manager'],
```
Comment (markdalgleish) [inline react/Autosuggest/Autosuggest.demo.js]: "UX" isn't a job title 😜

Comment (kosanna) [inline react/Autosuggest/Autosuggest.demo.js]: I'm happy to put there whatever :) `apple banana melon watermelon` :D

File: react/index.js [hunk 4]
```diff
@@ -7,6 +7,7 @@ export { default as Rating } from './Rating/Rating';
 export { default as Header } from './Header/Header';
 export { default as Footer } from './Footer/Footer';
 export { default as Loader } from './Loader/Loader';
+export { default as DisableScrollOutside } from './DisableScrollOutside/DisableScrollOutside';
```
Comment (markdalgleish) [inline react/index.js]: Not sure this should be exported, seems more like a utility than a company specific style.

Comment (kosanna) [inline react/index.js]: classifications :trollface:

**Comment by kosanna** [comment]
Closing for now as has issues

**Pull request closed without merging**